modules/cli/src/main/scala/scala/cli/commands/bsp/Bsp.scala

```diff
@@ -51,6 +53,24 @@ object Bsp extends ScalaCommand[BspOptions] {
   override def sharedOptions(options: BspOptions): Option[SharedOptions] =
     Option(latestSharedOptions(options))
 
+  private def refreshPowerMode(
+    latestLauncherOptions: LauncherOptions,
+    latestSharedOptions: SharedOptions,
+    latestEnvs: Map[String, String]
+  ): Unit = {
+    val previousPowerMode = ScalaCli.allowRestrictedFeatures
+    val configPowerMode = ConfigDbUtils.getLatestConfigDbOpt(latestSharedOptions.logger)
+      .flatMap(_.get(Keys.power).toOption)
+      .flatten
+      .getOrElse(false)
```

Comment:
Repeated logic with what is in `isSipScala`?

Reply:
The main difference is that this is called multiple times while the BSP server is up, and we care about getting the most recent version of `configDb`, which is why we're using `ConfigDbUtils.getLatestConfigDbOpt`, rather than relying on the lazy value.